platform/parsers/elastic_query_dsl/query_parser.go

```diff
@@ -579,7 +579,8 @@ func (cw *ClickhouseQueryTranslator) parseMatch(queryMap QueryMap, matchPhrase b
 					computedIdMatchingQuery := cw.parseIds(QueryMap{"values": []interface{}{subQuery}})
 					statements = append(statements, computedIdMatchingQuery.WhereClause)
 				} else {
-					simpleStat := model.NewInfixExpr(model.NewColumnRef(fieldName), model.MatchOperator, model.NewLiteral("'"+subQuery+"'"))
+					fullLiteral := model.NewLiteralWithEscapeType("'"+subQuery+"'", model.NotEscapedLikeFull)
```

Comment:
It looks like we are inconsistent with the way we do escape.  File above:
1.  File above: `fmt.Sprintf("'%s'", termAsStringToClickhouse)`
2. Here: `"'"+subQuery+"'"`

I would consider unifying or having a helper function.